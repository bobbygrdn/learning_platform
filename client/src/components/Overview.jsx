import React from 'react'

export default function Overview() {
    // TODO: Finish overview component
    return (
        <div>

        </div>
    )
}
