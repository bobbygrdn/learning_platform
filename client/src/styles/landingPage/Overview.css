.overviewText {
    padding: 0 1em;
    text-align: left;
    margin-bottom: 5em;
    pointer-events: none;
}

.callStatement {
    text-align: center;
    font-size: 26px;
    font-weight: bold;
    margin-bottom: 1em;
}

@media screen and (min-width: 1000px) {
    .overviewText {
        font-size: 22px;
        padding: 0em 4em;
    }
}