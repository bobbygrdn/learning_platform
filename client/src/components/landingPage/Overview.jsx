import React from 'react';
import '../../styles/landingPage/Overview.css';

export default function Overview() {
    return (
        <div className='overview' data-testid="overview">
            <p className='callStatement' data-testid="callStatement">Welcome to Valor Academy!</p>
            <p className='overviewText' data-testid="overviewText">
                This platform was created to engage and help guide the developer community into a new style of learning. Our focus is to allow developers in all stages of their journey to grow and become more confident in their abilities. As you progress through the different courses, lessons, quizzes and more, you will see that we focus on providing great content and a roadmap for you to level up!
            </p>
        </div>
    )
}
