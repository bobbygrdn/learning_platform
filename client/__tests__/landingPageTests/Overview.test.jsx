import React from 'react';
import { render, screen } from '@testing-library/react';
import Overview from '../../src/components/landingPage/Overview';

it('renders without crashing', () => {
    const { container } = render(<Overview />);
    expect(container).toMatchSnapshot();
})

it('should render paragraph element with class of text', () => {
    render(<Overview />);
    expect(screen.getAllByTestId('overviewText')).toHaveLength(1);
})

it('should render paragraph element with class of callStatement', () => {
    render(<Overview />);
    expect(screen.getAllByTestId('callStatement')).toHaveLength(1);
})