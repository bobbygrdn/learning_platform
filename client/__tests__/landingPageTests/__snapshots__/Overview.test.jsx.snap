// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="overview"
    data-testid="overview"
  >
    <p
      class="callStatement"
      data-testid="callStatement"
    >
      Welcome to Valor Academy!
    </p>
    <p
      class="overviewText"
      data-testid="overviewText"
    >
      This platform was created to engage and help guide the developer community into a new style of learning. Our focus is to allow developers in all stages of their journey to grow and become more confident in their abilities. As you progress through the different courses, lessons, quizzes and more, you will see that we focus on providing great content and a roadmap for you to level up!
    </p>
  </div>
</div>
`;
